server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java

```diff
@@ -163,6 +164,47 @@ protected boolean doStart(JobContext context) throws Exception {
         return true;
     }
 
+    private List<OffsetString> getSplitSql() {
+        SqlCommentProcessor processor = ConnectionSessionUtil.getSqlCommentProcessor(connectionSession);
+        StringBuffer buffer = new StringBuffer();
+        List<OffsetString> splitSqlList = new ArrayList<>();
+
+        if (CollectionUtils.isNullOrEmpty(parameters.getSqlObjectIds())) {
+            String sqlContent = parameters.getSqlContent();
+            log.info("SQL content={}", sqlContent);
+
+            List<OffsetString> offsetStringList = processor.split(buffer, sqlContent);
+            result.setTotalStatements(offsetStringList.size());
```

Comment:
result.setTotalStatements may set out of getSplitSql method, responsibility should isolated